src/wallet/walletdb.cpp

```diff
@@ -465,6 +477,39 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
 
             pwallet->LoadKeyPool(nIndex, keypool);
         }
+        else if (strType == "pending_firstupdate")
+        {
+            std::string strName, strJsonData;
+            ssKey >> strName;
+            ssValue >> strJsonData;
+            UniValue jsonData, v1, v2, v3;
+            std::string txid, rand, pendingData;
+            jsonData.read(strJsonData);
+
+            v1 = find_value (jsonData, "txid");
+            txid = v1.get_str();
+
+            v2 = find_value (jsonData, "rand");
+            rand = v2.get_str();
+
+            v3 = find_value (jsonData, "data");
+            pendingData = v3.get_str();
+
+            if(v1.type() != UniValue::VSTR ||
+               v2.type() != UniValue::VSTR ||
+               v3.type() != UniValue::VSTR) {
+                strErr = strprintf("Bad data while importing pending name firstupdate: %s\n", strName.c_str());
+                return false;
+            }
+
+            NameNewReturn ret;
+            ret.hex = txid;
+            ret.rand = rand;
+            ret.data = pendingData;
```

Comment:
This has me wonder about the serialisation of `NameNewReturn`.  You could actually define "proper" serialisation of the struct in the same way that other stuff in the code is serialised (e. g., transactions).  That would fit better into the existing code here than using JSON.

Reply:
I agree and I'll add this to the list.